radix-engine/src/system/system_callback.rs

```diff
@@ -113,42 +150,110 @@ pub struct System<C: SystemCallbackObject> {
 impl<C: SystemCallbackObject> KernelCallbackObject for System<C> {
     type LockData = SystemLockData;
     type CallFrameData = Actor;
-    type InitInput = C::InitInput;
+    type InitInput = SystemInit<C::InitInput>;
 
     fn init<S: BootStore>(
         store: &S,
         executable: &Executable,
-        execution_config: &ExecutionConfig,
-        init_input: C::InitInput,
+        init_input: SystemInit<C::InitInput>,
     ) -> Result<Self, BootloadingError> {
-        let costing_parameters = {
+        let mut system_parameters = {
             let system_boot = store
                 .read_substate(
                     TRANSACTION_TRACKER.as_node_id(),
                     BOOT_LOADER_PARTITION,
                     &SubstateKey::Field(BOOT_LOADER_SYSTEM_SUBSTATE_FIELD_KEY),
                 )
                 .map(|v| scrypto_decode(v.as_slice()).unwrap())
-                .unwrap_or(SystemBoot::V1 {
-                    costing_parameters: CostingParameters::default(),
-                });
+                .unwrap_or(SystemBoot::V1(SystemParameters::default()));
 
             match system_boot {
-                SystemBoot::V1 { costing_parameters } => costing_parameters,
+                SystemBoot::V1(system_parameters) => system_parameters,
             }
         };
-        let callback = C::init(store, init_input)?;
 
-        let mut modules = SystemModuleMixer::new(
-            execution_config.enabled_modules,
-            execution_config.network_definition.clone(),
+        let callback = C::init(store, init_input.callback_init)?;
+
+        let enabled_modules = {
+            let mut enabled_modules = EnabledModules::AUTH | EnabledModules::TRANSACTION_RUNTIME;
+            if !executable.is_system() {
+                enabled_modules |= EnabledModules::LIMITS;
+                enabled_modules |= EnabledModules::COSTING;
+            };
+
+            if init_input.enable_kernel_trace {
+                enabled_modules |= EnabledModules::KERNEL_TRACE;
+            }
+            if init_input.execution_trace.is_some() {
+                enabled_modules |= EnabledModules::EXECUTION_TRACE;
+            }
+
+            if let Some(system_overrides) = &init_input.system_overrides {
```

Comment:
We can actually use `.remove()`, which improves readability. Would also consolidate moving this block down to be together with other overrides.

Reply:
Good point on `.remove()`. I did have this originally consolidated in the block below but thought it was clearer that `enabled_modules` could be described immutably in this manner. I'm good either way though.